stock_restrict_by_planned_consumed_date/models/stock_quant.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2024 Foodles (https://www.foodles.co)
+# @author Pierre Verkest <pierreverkest84@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, models
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+
+    @api.model
+    def _gather(self, *args, **kwargs):
+        quants = super()._gather(*args, **kwargs)
+        if planned_consumed_date := self.env.context.get(
+            "restrict_planned_consumed_date"
+        ):
+            quants = quants.filtered(
+                lambda quant, planned=planned_consumed_date: quant.lot_id is False
+                or quant.lot_id
+                and quant.lot_id.expiration_date > planned
```

Comment:
**suggestion,nitpick:** Maybe add parenthesis to clearly separate the `or` and the `and`?